app/models/storage_hosts_wwpn_candidate.rb

```diff
@@ -0,0 +1,27 @@
+class StorageHostsWwpnCandidate < ApplicationRecord
+  belongs_to :ext_management_system, :foreign_key => :ems_id
+  belongs_to :physical_storage, :inverse_of => :storage_hosts_wwpn_candidates
+
+  def self.class_by_ems(ext_management_system)
+    ext_management_system && ext_management_system.class::StorageHostsWwpnCandidate
```

Comment:
There is no `:type` column in your schema PR so this won't work, do you actually need to subclass this?  I don't see a subclass in your provider PR.